sklearn/model_selection/tests/test_search.py

```diff
@@ -641,6 +641,69 @@ def test_refit():
     clf.fit(X, y)
 
 
+def test_refit_callable():
+    """
+    Test refit=callable, which adds flexibility in identifying the
+    "best" estimator.
+    """
+    def refit_callable(cv_results):
+        """
+        A dummy function tests `refit=callable` interface.
+        Return the index of a model that has the least
+        `mean_test_score`.
+        """
+        # Fit a dummy clf with `refit=True` to get a list of keys in
+        # clf.cv_results_.
+        X, y = make_classification(n_samples=100, n_features=4,
+                                   random_state=42)
+        clf = GridSearchCV(LinearSVC(random_state=42), {'C': [0.01, 0.1, 1]},
+                           scoring='precision', refit=True)
+        clf.fit(X, y)
+        # Ensure that `best_index_ != 0` for this dummy clf
+        assert clf.best_index_ != 0
```

Comment:
Why can't best_index not be 0? What if the first option in the hyperparameter grid is in fact the best option?

Reply:
This assertion tries to prove that `refit=refit_callable` and `refit=True` have different behaviors on the same data.

Reply:
`refit=callable` selected the index with min test score whereas `refit=True` selected index with the min `rank_test_score`.

Reply:
There is a comment above to explain